src/hotspot/share/opto/node.cpp

```diff
@@ -2236,17 +2230,27 @@ void Node::verify(Node* n, int verify_depth) {
       assert(C->cached_top_node() == n, "TOP node must be unique");
     }
 
-    for (uint i = 0; i < n->len(); i++) {
-      Node* x = n->in(i);
+    uint in_len = n->len();
+    for (uint i = 0; i < in_len; i++) {
+      Node* x = n->_in[i];
       if (!x || x->is_top()) {
         continue;
       }
 
       // Verify my input has a def-use edge to me
       // Count use-def edges from n to x
-      int cnt = 0;
-      for (uint j = 0; j < n->len(); j++) {
-        if (n->in(j) == x) {
+      int cnt = 1;
+      for (uint j = 0; j < i; j++) {
+        if (n->_in[j] == x) {
+          cnt++;
+          break;
+        }
+      }
+      if (cnt == 2) {
```

Comment:
I think it should be `cnt > 1`.

Reply:
previous loop breaks when meet first input which is same with x, so cnt must be 2 if x is duplicated with previous input.

Reply:
You are right. I missed `break`. Using an other local VectorSet here may not be cheeper. Okay.
Add comment here. Something about that we processed this node already.

Reply:
Thanks for your review! comment is added.